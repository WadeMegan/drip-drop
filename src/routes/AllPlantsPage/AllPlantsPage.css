
.available-plants-section h2{
  text-align: center;
  padding-bottom: 15px;
}
  
  .all-plants-container{
    width:100%;
    padding:20px;
    display:flex;
    flex-direction: column;
    justify-content:center;
    align-items:center;
  }
  
  .directions {
    color:grey;
  }
  
  .directions a {
    color: black;
  }
  
  .directions a:hover {
    color: grey;
    cursor:pointer;
  }
  
  #search-plants{
    border:1px solid grey;
  }
  

.switchPagesButton{
  background-color: #f5d60d;
  color:black;
  border-radius: 25px;
  width:100%;
  height:50px;
  color:#041f3c;
  display:flex;
}

.pageButton{
  width:50%;
  display:flex;
  align-items:center;
  justify-content: center;
}

.notCurrent:hover{
background-color:#e6c90d;
cursor: pointer;
}

.pageButton p, .pageButton{
  text-decoration: none;
  color:#041f3c;
}


.availableButton{
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
}

.yourPlantsButton{
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
}

.current{
  -webkit-box-shadow: inset 1px 1px 10px #d8bd0e;
  -moz-box-shadow:    inset 1px 1px 10px #d8bd0e;
  box-shadow:         inset 1px 1px 10px #d8bd0e;
}

.banner{
  background: url('../../images/waterBanner.jpg') no-repeat bottom left;
  background-size:100%;
  background-repeat:no-repeat;
  height:150px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items:center;
  padding:20px;
}