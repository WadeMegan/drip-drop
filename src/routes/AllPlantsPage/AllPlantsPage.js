import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './AllPlantsPage.css'
import PlantItem from '../../components/PlantItem/PlantItem'
import PlantListContext from '../../contexts/PlantListContext'
import PlantApiService from '../../services/plant-api-services'
import UserService from '../../services/user-service'
import Error from '../../components/Error/Error'

export default class AllPlantsPage extends Component {
    static contextType = PlantListContext

    state = {
        isFiltered:false,
        filteredPlants: []
    }

    componentDidMount=()=>{
        this.context.clearError()
        const userId = UserService.getUserToken()

        if(!this.context.usersPlants.length){
            PlantApiService.getUsersPlants(userId)
                .then(this.context.setUsersPlants)
                .catch(this.context.setError)
        }
        
        if(!this.context.plantList.length){
            PlantApiService.getAllPlants()
                .then(this.context.setPlantList)
                .catch(this.context.setError)
        }
    }


    renderPlants=(plants)=>{
        return plants.map(plant =>
            <PlantItem key={plant.id} plant={plant}/>
        ) 
    }

    filterPlants = (event) => {

        let clonedArray = JSON.parse(JSON.stringify(this.context.plantList))

        let plants = clonedArray.filter((plant)=>{
            return plant.name.toLowerCase().search(event.target.value.toLowerCase()) !== -1
        })

        this.setState({
            isFiltered:true,
            filteredPlants: plants
        })

    }

    render(){
        return (
            <Error>
                <div className='switchPagesNav'>
                    <Link className='pageButton directionsButton notCurrent' to='directions'>
                        <p className='buttonItem'>Directions</p>
                    </Link>
                    <div className='pageButton availableButton current'>
                        <p className='buttonItem'>Available Plants</p>
                    </div>
                    <Link className='pageButton yourPlantsButton notCurrent' to='your-plants'>
                        Your Plants
                    </Link>
                </div>
                <section className='availablePlantsSection'>
                    <div className='banner'>
                        <h2>Available Plants</h2>
                        <form className='searchForm'> 
                            <label htmlFor='searchPlants'>Search all plants: </label>
                            <input type='text' name='searchPlants' id='searchPlants' placeholder='zz plant' onChange={this.filterPlants}/>
                        </form>
                    </div>
                    <div className='allPlantsContainer'>
                        {this.state.isFiltered
                        ? this.renderPlants(this.state.filteredPlants)
                        : this.renderPlants(this.context.plantList)}
                    </div>
                </section>
            </Error>
        )
    }
}

