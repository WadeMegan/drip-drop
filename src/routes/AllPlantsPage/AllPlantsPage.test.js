import React from 'react'
import ReactDOM from 'react-dom'
import AllPlantsPage from './AllPlantsPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><AllPlantsPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AllPlantsPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
