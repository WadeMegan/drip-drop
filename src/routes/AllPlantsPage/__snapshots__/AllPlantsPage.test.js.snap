// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
Array [
  <div
    className="switchPagesNav"
  >
    <a
      className="pageButton directionsButton notCurrent"
      href="/directions"
      onClick={[Function]}
    >
      <p
        className="buttonItem"
      >
        Directions
      </p>
    </a>
    <div
      className="pageButton availableButton current"
    >
      <p
        className="buttonItem"
      >
        Available Plants
      </p>
    </div>
    <a
      className="pageButton yourPlantsButton notCurrent"
      href="/your-plants"
      onClick={[Function]}
    >
      Your Plants
    </a>
  </div>,
  <section
    className="availablePlantsSection"
  >
    <div
      className="banner"
    >
      <h2>
        Available Plants
      </h2>
      <form
        className="searchForm"
      >
        <label
          htmlFor="searchPlants"
        >
          Search all plants: 
        </label>
        <input
          id="searchPlants"
          name="searchPlants"
          onChange={[Function]}
          placeholder="zz plant"
          type="text"
        />
      </form>
    </div>
    <div
      className="allPlantsContainer"
    />
  </section>,
]
`;
